src/librustdoc/clean/types.rs

```diff
@@ -648,12 +647,38 @@ impl<'a> Iterator for ListAttributesIter<'a> {
 crate trait AttributesExt {
     /// Finds an attribute as List and returns the list of attributes nested inside.
     fn lists(&self, name: Symbol) -> ListAttributesIter<'_>;
+    
+    fn span(&self) -> Option<rustc_span::Span>;
+    
+    fn inner_docs(&self) -> bool;
+    
+    fn other_attrs(&self) -> Vec<ast::Attribute>;
 }
 
 impl AttributesExt for [ast::Attribute] {
     fn lists(&self, name: Symbol) -> ListAttributesIter<'_> {
         ListAttributesIter { attrs: self.iter(), current_list: Vec::new().into_iter(), name }
     }
+    
+    fn span(&self) -> Option<rustc_span::Span> {
+        self.iter()
+            .filter(|attr| attr.doc_str().is_some())
+            .map(|attr| attr.span)
+            .next()
+    }
+
+    fn inner_docs(&self) -> bool {
+        self.iter()
+        .find(|a| a.doc_str().is_some())
+        .map_or(true, |a| a.style == AttrStyle::Inner)
+    }
+
+    fn other_attrs(&self) -> Vec<ast::Attribute> {
+        self.iter().filter_map(|attr| match attr.doc_str() {
+            Some(_) => None,
+            None => Some(attr.clone()),
+        }).collect()
+    }
```

Comment:
Hmm, you added this function, but didn't remove the `other_attrs` field - was that intentional?

Reply:
This implementation was required to remove `cfg` from `Attributes` but to be able to entirely remove the `other_attrs` field from `Attributes` we will have to move it in `Item`. It will be done in a future MR.